buildconfig/Jenkins/Conda/conda-buildscript-linux

```diff
@@ -0,0 +1,131 @@
+
+# SCRIPT_DIR discovery from https://stackoverflow.com/a/9107028 by https://stackoverflow.com/users/1184238/andrew-norrie
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+WORKSPACE=$1
+shift
+CMAKE_PRESET=$1
+shift
+EXPECTED_MAMBAFORGE_PATH=$WORKSPACE/mambaforge # Install into the WORKSPACE_DIR
+EXPECTED_CONDA_PATH=$EXPECTED_MAMBAFORGE_PATH/bin/conda
+CONDA_ENV_NAME="mantid-developer"
+CPU_COUNT="$(grep -c ^processor /proc/cpuinfo)" # Not physical but logical cores, includes hyper threaded.
+XVFB_SERVER_NUM=101
+BUILD_DIR=$WORKSPACE/build
+
+ENABLE_BUILD_CODE=true
+ENABLE_UNIT_TESTS=true
+ENABLE_SYSTEM_TESTS=false
+ENABLE_PACKAGE=false
+ENABLE_DOCS=false
+ENABLE_DOC_TESTS=false
+ENABLE_DEV_DOCS=false
+CLEAN_BUILD=false
+CLEAN_EXTERNAL_PROJECTS=false
+USE_CORE_DUMPS=false
+EXTRA_CMAKE_FLAGS=""
+
+# Handle flag inputs
+while [ ! $# -eq 0 ]
+do
+    case "$1" in
+        --enable-systemtests) ENABLE_SYSTEM_TESTS=true ;;
+        --enable-package) ENABLE_PACKAGE=true ;;
+        --enable-docs) ENABLE_DOCS=true ;;
+        --enable-doctests) ENABLE_DOC_TESTS=true ;;
+        --enable-dev-docs) ENABLE_DEV_DOCS=true ;;
+        --clean-build) CLEAN_BUILD=true ;;
+        --clean-external-projects) CLEAN_EXTERNAL_PROJECTS=true ;;
+        --use-core-dumps) USE_CORE_DUMPS=true ;;
+        --extra-cmake-flags)
+            EXTRA_CMAKE_FLAGS="$2"
+            shift
+            ;;
+        *)
+            echo "Argument not accepted: $1"
+            exit 1
+            ;;
+  esac
+  shift
+done
+
+# Setup Mambaforge
+$SCRIPT_DIR/download-and-install-mambaforge $EXPECTED_MAMBAFORGE_PATH $EXPECTED_CONDA_PATH $CLEAN_BUILD
+
+# Setup Conda environment
+$SCRIPT_DIR/setup-conda-env $WORKSPACE $EXPECTED_CONDA_PATH $CONDA_ENV_NAME
+
+# Activate Conda environment
+. $WORKSPACE/mambaforge/etc/profile.d/conda.sh
+conda activate $CONDA_ENV_NAME
+
+# Clean the source tree to remove stale configured files ignoring the build directory
+git clean -d -x --force --exclude=build --exclude=mambaforge --exclude=".Xauthority-*"
+
+# Clean up build folder
+if [[ $CLEAN_BUILD  == true ]]; then
+    rm -rf $BUILD_DIR
+fi
+
+mkdir -p $BUILD_DIR
+
+rm -rf ${BUILD_DIR}/bin ${BUILD_DIR}/ExternalData ${BUILD_DIR}/Testing
+find ${BUILD_DIR} \( -name 'TEST-*.xml' -o -name 'Test.xml' \) -delete
+
+if [[ ${CLEAN_EXTERNAL_PROJECTS} == true ]]; then
+    rm -rf $BUILD_DIR/eigen-*
+    rm -rf $BUILD_DIR/googletest-*
+fi
+
+# Clean up artifacts from previous builds
+rm -f -- *.dmg *.rpm *.deb *.tar.gz *.tar.xz
+
+# Should we even build the code?
+if ${SCRIPT_DIR}/../check_for_changes dev-docs-only || ${SCRIPT_DIR}/../check_for_changes user-docs-only; then
+    ENABLE_BUILD_CODE=false
+    ENABLE_UNIT_TESTS=false
+    ENABLE_SYSTEM_TESTS=false
+fi
+
+# Run the script that handles the actual building
+$SCRIPT_DIR/build $WORKSPACE $CMAKE_PRESET $ENABLE_DOCS $ENABLE_DEV_DOCS $ENABLE_BUILD_CODE $ENABLE_UNIT_TESTS $ENABLE_SYSTEM_TESTS $EXTRA_CMAKE_FLAGS
+
+# Clean up prior to testing
```

Comment:
How about we extract this out to a separate step script too? Something like `prepare-for-testing`  that does up to line 128?

Reply:
I was trying to limit the amount of files we ended up calling as it can make it harder to find exactly where something happens and add more complexity than it should.

Perhaps it should just go in the `run-tests` script?

Reply:
That's fine by me also.
For readability it's good to keep things at the same level of abstraction so we have 2 external script calls that are not punctuated by lower level detail.